components/opc-ua/src/Microsoft.Azure.IIoT.OpcUa.Protocol/src/Services/SubscriptionServices.cs

```diff
@@ -763,15 +765,34 @@ internal sealed class SubscriptionWrapper : ISubscription {
                                 continue;
                             }
 
-                            var monitoredItemNotification = notification.Events[i]
-                                .ToMonitoredItemNotification(monitoredItem);
-                            if (message == null) {
-                                continue;
-                            }
-                            message.Notifications?.Add(monitoredItemNotification);
-
                             if (monitoredItem.Handle is MonitoredItemWrapper itemWrapper) {
                                 var pendingAlarmsOptions = itemWrapper?.EventTemplate?.PendingAlarms;
+                                var evFilter = (itemWrapper.Item.Filter as EventFilter);
+                                var eventTypeIndex = evFilter.SelectClauses.IndexOf(
+                                    evFilter.SelectClauses
+                                        .FirstOrDefault(x => x.TypeDefinitionId == ObjectTypeIds.BaseEventType && x.BrowsePath?.FirstOrDefault() == "EventType"));
+
+                                // now, is this a regular event or RefreshStartEventType/RefreshEndEventType?
+                                if (eventTypeIndex != -1) {
```

Comment:
What happens, if I select the events RefreshStartEventType/RefreshEndEventType for custom usage? Is then the PendingAlarms is not cleared?

Reply:
If you haven't set up BaseEventType/EventType in the select clause, the code will add it internally.